lib/facade/MonitoringFacade.ts

```diff
@@ -124,15 +137,42 @@ export class MonitoringFacade extends MonitoringScope {
   protected readonly dashboardFactory?: IDashboardFactory;
   protected readonly createdSegments: IDashboardSegment[];
 
-  constructor(scope: Construct, id: string, props: MonitoringFacadeProps) {
+  constructor(scope: Construct, id: string, props?: MonitoringFacadeProps) {
     super(scope, id);
 
-    this.metricFactoryDefaults = props.metricFactoryDefaults;
-    this.alarmFactoryDefaults = props.alarmFactoryDefaults;
-    this.dashboardFactory = props.dashboardFactory;
+    this.metricFactoryDefaults =
+      props?.metricFactoryDefaults ??
+      MonitoringFacade.getDefaultMetricFactoryDefaults();
+    this.alarmFactoryDefaults =
+      props?.alarmFactoryDefaults ??
+      MonitoringFacade.getDefaultAlarmFactoryDefaults(id);
+    this.dashboardFactory =
+      props?.dashboardFactory ??
+      MonitoringFacade.getDefaultDashboardFactory(this, id);
+
     this.createdSegments = [];
   }
 
+  private static getDefaultMetricFactoryDefaults() {
+    return {};
+  }
```

Comment:
Let's add specific return type of `MetricFactoryDefaults` to this method. (same for the next 2)